nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -115,7 +312,26 @@ public void removeInspectionHook(String inspectionHookId) throws Exception {
             return;
         }
 
-        this.utils.removeSingleInspectionHook(inspectionHookId);
+        FlowClassifier flowClassifier = this.osCalls.getFlowClassifier(inspectionHookId);
+        if (flowClassifier == null) {
+            LOG.warn("Flow Classifier {} does not exist on openstack", inspectionHookId);
+            return;
+        }
+
+        PortChain portChain = this.utils.fetchContainingPortChainForFC(flowClassifier.getId());
```

Comment:
when you delete a flow classifier, this should happen automatically right? Can you please check?

ideally this would only be the code below and handling 404 to ignore and propogate all other errors

`this.osCalls.deleteFlowClassifier(flowClassifier.getId())`

Reply:
I commented out that line and the if statement following. (essentially the call to `this.osCalls.updatePortChain()` ). I got got this:

```ERROR org.osc.controller.nsfc.api.NeutronSfcSdnRedirectionApi - Error removing flow classifier 35ee6b29-103a-4200-aa3c-9e3173047530. Response 409 (Flow Classifier 35ee6b29-103a-4200-aa3c-9e3173047530 in use.)```